docs: remove version from plugin documentation howto (#831)
canonical/craft-parts: docs/how-to/document-plugin.rst

It is also useful to indicate if the plugin replaces an existing plugin, or
should be used instead of another similar plugin.

> If documenting an application-specific plugin, it may be appropriate to include

Reviewer: How about:

```suggestion
If you're documenting an application-specific plugin, we encourage you to include
```

Author: It's not necessarily encouraged though.

Reviewer: Oh, I thought this meant that non-application-specific plugins should never have versions, but some application-specific ones should. What are we advising for other plugins?

Author: For generic craft-parts plugins, they should not have a version. For application-specific ones, it depends on the app. If the app uses versioned documentation, it's not necessary. If the app's documentation is unversioned, it might make sense.

Reviewer: What do you think about including all that nuance? I'm not sure I could grok that from this template.

Author: That seems more like something that's up to the app development team.

Author: I've removed the paragraph as a whole.

> If documenting an application-specific plugin, it may be appropriate to include
> the version of the application in which this plugin first appears here.
> 

Author: ```suggestion
```
